src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -242,17 +244,21 @@ public function set($id, $service, $scope = self::SCOPE_CONTAINER)
      */
     public function has($id)
     {
-        $id = strtolower($id);
-
-        if ('service_container' === $id) {
-            return true;
+        for ($i = 2;;) {
+            if ('service_container' === $id
+                || isset($this->services[$id])
+                || isset($this->aliases[$id])
+                || array_key_exists($id, $this->services)
+                || method_exists($this, 'get'.strtr($id, $this->underscoreMap).'Service')
```

Comment:
I suggest checking the methodMap before doing the replacements here (as done in ``get``)